src/hooks/useClickOutside.tsx

```
@@ -0,0 +1,22 @@
+/* eslint-disable n/no-callback-literal */
+import { type RefObject, useEffect } from "react";
+
+export default function useClickOutside(
+  ref: RefObject<HTMLDivElement>,
+  callback: (el: boolean) => void
+): void {
+  useEffect(() => {
+    function handleClickOutside(event: Event): void {
+      const { target } = event;
+      if (target !== null) {
+        if (ref.current != null && !ref.current.contains(target as Node)) {
+          callback(false);
+        }
+      }
+    }
```

I am not sure, but will we benefit from taking it out of the `useEffect` and wrapping it with `useCallback` to track for callback and ref?
Also, see the above comment about the potential to move it to the component.

Also, I think the callback pattern should be like this: `callback(err, false);`

But you can suppress the error with:
```
const isOutside = false;
callback(isOutside);
```
if the first error argument is not provided.

---

To this and comment above:
I actually wanted it to be retrieved as a hook exactly cause we might use that later with any pop-ups, dropdowns and etc. I think it's pretty convenient to use, so why not move it to a hook straight away.

Yes, moving to callback might benefit, you're right, will do that